### ISSUE TITLE
Controlling the order of completion groups?

### ISSUE THREAD

**Issue by alphapapa** [minad/consult#624]
Hi Daniel,

Apologies if this is not in Consult's domain; I looked at some of Consult's source code, and checked the Elisp manual, but I couldn't find anything about this.

I've been experimenting with `consult--multi` to develop a counterpart to my Helm-based "find-everything" command that combines sources of various types, and I'm very pleased except for one thing: the order in which completion groups are displayed seems to have no relation to the order in which the sources are passed to `consult--multi`.  For example:

```el
(defun ap/consult-everything ()
  (interactive)
  ;; The order of the sources when shown doesn't seem to match the
  ;; order of them here.
  (consult--multi '(ap/consult-source-bufler
                    consult--source-bookmark
                    consult--source-recent-file
                    ;; consult--source-buffer
                    ap/consult-source-org-recent-headings
                    consult--source-project-recent-file
                    consult--source-project-buffer
                    ap/consult-source-org-ql-views)))
```
When I run that command, I'd like the items from the Bufler source to be shown first, but instead the `consult--source-bookmark` items are shown first.  The other groups also don't seem to be in any discernible order, neither alphabetical nor the order in the argument list.

Is this something that we can control, or does this need to be fixed in Emacs itself?

Thanks for all your work.

**Comment by minad** [comment]
Hey Adam,

try passing :sort=nil to preserve the original order (source and candidates). Otherwise the candidates will be sorted by the UI (e.g. vertico-sort-function). The order of the candidates determines the order of the groups. I hope this helps!

**Comment by alphapapa** [comment]
Thank you.  :)

**Comment by alphapapa** [comment]
By the way, @minad, what do you think about proposing (i.e. within Emacs itself) that completion groups be sortable separately from the items within them?

**Comment by minad** [comment]
@alphapapa I like the current way of sorting and don't see the need for an addition. Do you have a use case where this would be needed?

**Comment by alphapapa** [comment]
@minad Yes, for example, in my multi-source command like in the example I showed, I'd like files and buffers and such to be sorted by recency within each group, but I want the groups (i.e. sources) themselves to be in a specified order.  I don't want whichever group happens to have the most recent item to be sorted before the other groups, because it's not useful that way, never knowing what order the groups will be in.  The order of the groups as I listed them is the most useful to me, with the most important sources first.

This is easy with Helm, because I can pass multiple sources to Helm and they maintain the given order, with items within the sources being sorted as specified.
